Sources/NIOHTTP1Server/main.swift

```diff
@@ -41,8 +41,36 @@ private final class HTTPHandler: ChannelInboundHandler {
     public typealias InboundIn = HTTPServerRequestPart
     public typealias OutboundOut = HTTPServerResponsePart
 
+    private enum State {
+        case idle
+        case waitingForRequestBody
+        case sendingResponse
+
+        mutating func requestReceived() {
+            guard case .idle = self else {
+                fatalError("Invalid state for request received: \(self)")
```

Comment:
I think you could simplify this by just using a `precondition`.

Something like:

```swift
precondition(self == .idle, "Invalid......")
```

Same for other places as well